model/CSVDialog.cc

```diff
@@ -34,6 +53,111 @@ void CSVDialog::reportFromFile(const std::string& input, const std::string& outp
   reportFromCSVFile(is,of,spec);
 }
 
+// Return file name after downloading a CSV file from the web.
+std::string CSVDialog::loadWebFile(const std::string& url)
+{
+    try
+    {
+		// Parse input URL
+        boost::regex ex("(http|https)://([^/ :]+):?([^/ ]*)(/?[^ #?]*)\\x3f?([^ #]*)#?([^ ]*)");
+        boost::cmatch what;
+        regex_match(url.c_str(), what, ex);
+        
+        // The web domain 
+        auto const host = what[2];
+        
+        // Set the port type
+        string port = string(what[1].first, what[1].second);
+        if (string(what[1].first, what[1].second)=="https") port = "https"; 
```

Comment:
I believe what[1] is directly convertible/comparable to string. If not (sometimes its not), then what.str(1) certainly is.

Also, this code block is more simply written port=what[1], as it can only be http or https.

Also, you have tested the return value of regex_match for success.

A better term for "port" is "protocol". The actual port value is given by what[3]. If what[3] is empty, the port = 80 in the case of http and 443 in the case of https.

Reply:
I think I've fixed all of this now. I found the regex expressions at these two pages:

[https://stackoverflow.com/questions/2616011/easy-way-to-parse-a-url-in-c-cross-platform](
https://stackoverflow.com/questions/2616011/easy-way-to-parse-a-url-in-c-cross-platform)

[http://www.zedwood.com/article/cpp-boost-url-regex](http://www.zedwood.com/article/cpp-boost-url-regex)